sync-android/src/main/java/com/cloudant/sync/sqlite/android/AndroidSQLCipherSQLite.java

```diff
@@ -0,0 +1,206 @@
+/**
+ * Copyright (c) 2013 Cloudant, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.sync.sqlite.android;
+
+/**
+ * Similar to AndroidSQLite class.  Replaces Android SQLite implementation with
+ * SQLCipher-based SQLite.
+ * SQLCipher adds:
+ * - TODO A user-based password (provided with the KeyProvider object) for encryption
+ * - TODO Method to change password (will be handled through key management feature)
+ *
+ * Created by estebanmlaver.
+ */
+
+import com.cloudant.sync.datastore.encryption.KeyProvider;
+import com.cloudant.sync.datastore.encryption.KeyUtils;
+import com.cloudant.sync.sqlite.ContentValues;
+import com.cloudant.sync.sqlite.Cursor;
+import com.cloudant.sync.sqlite.SQLDatabase;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+import net.sqlcipher.database.SQLiteConstraintException;
+import net.sqlcipher.database.SQLiteDatabase;
+
+public class AndroidSQLCipherSQLite extends SQLDatabase {
+
+    SQLiteDatabase database = null;
+
+    /**
+     * Constructor for creating SQLCipher-based SQLite database.
+     * @param path full file path of the db file
+     * @param provider Provider object that contains the key to encrypt the SQLCipher database
+     * @return
+     */
+    public static AndroidSQLCipherSQLite createAndroidSQLite(String path, KeyProvider provider) {
+
+        //Call SQLCipher-based method for opening database, or creating if database not found
+        SQLiteDatabase db = SQLiteDatabase.openOrCreateDatabase(path,
```

Comment:
Does `openOrCreateDatabase ` check if the key is correct? In iOS, I have to execute `SELECT count(*) FROM sqlite_master;` to know if it works (more details [here](https://www.zetetic.net/sqlcipher/sqlcipher-api/#key)).